fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/KeyValueCursor.java

```diff
@@ -192,7 +192,15 @@ public static Builder withSubspace(@Nonnull Subspace subspace) {
             return new Builder(subspace);
         }
 
-        public KeyValueCursor build() {
+        public KeyValueCursor build() throws RecordCoreException {
+            if (subspace == null) {
+                throw new RecordCoreException("record subspace must be supplied");
+            }
+
+            if (context == null) {
+                throw new RecordCoreException("record context must be supplied");
+            }
+
```

Comment:
It looks like this is still missing a check to make sure that the `ScanProperties` isn't `null`.

Reply:
DONE:
- Check for scanProperties added.
- Unit tests added.